src/hotspot/share/runtime/arguments.cpp

```diff
@@ -2653,20 +2655,28 @@ jint Arguments::parse_each_vm_init_arg(const JavaVMInitArgs* args, bool* patch_m
           set_mode_flags(_comp);
     // -Xshare:dump
     } else if (match_option(option, "-Xshare:dump")) {
+#if INCLUDE_CDS
       DumpSharedSpaces = true;
+#endif
     // -Xshare:on
     } else if (match_option(option, "-Xshare:on")) {
+#if INCLUDE_CDS
       UseSharedSpaces = true;
       RequireSharedSpaces = true;
+#endif
     // -Xshare:auto || -XX:ArchiveClassesAtExit=<archive file>
     } else if (match_option(option, "-Xshare:auto")) {
+#if INCLUDE_CDS
       UseSharedSpaces = true;
       RequireSharedSpaces = false;
+#endif
       xshare_auto_cmd_line = true;
     // -Xshare:off
     } else if (match_option(option, "-Xshare:off")) {
```

Comment:
These flags should all immediately be rejected if used in a non-CDS build. Yes this is pre-existing but it now looks totally broken with the new changes. There should be something like:
```
#ifndef INCLUDE_CDS
} else if (match_option(option, "-Xshare")) {  // or whatever we need to match the prefix only
  warning("Option %s is not supported in this VM", option);
#else 
// process flags as usual
#endif
```

Reply:
Hi David,  I adjusted the coding following your request (however we need to use `#if`  not  `#ifndef`   here).